src/diagram/components/dialog/dialog-row.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import classnames from "classnames";
+
+import "./dialog.scss";
+
+interface ITextArea {
+  cols: number;
+  rows: number;
+}
+interface IDialogRow {
+  invalid?: boolean;
+  label: string;
+  maxCharacters?: number;
+  setValue: (value: string) => void;
+  textarea?: ITextArea; // Include { cols, rows } to render a textarea instead of text input
+  value: string;
+  width?: number; // Width of text input
+}
+export const DialogRow = ({ invalid, label, maxCharacters, setValue, textarea, value, width }: IDialogRow) => {
+  const style = { width };
+  const classes = classnames("dialog-input", { invalid }, textarea ? "dialog-textarea" : "dialog-text");
+  return (
+    <div className="dialog-row">
+      <label className="dialog-label" htmlFor={label}>
+        {label}
+      </label>
+      { textarea
+        ? <textarea
+          className={classes}
+          id={label}
```

Comment:
Given that `id`s are required to be unique throughout the page, I worry that using the label as the `id` could result in `id` conflicts, given that there's nothing to prevent other items on the page from having an `id` like `Name` or `Notes`.

Reply:
I would also be worried about this, but who else would be crazy enough to use a generic id like Name or Notes? ;)